sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -544,6 +544,15 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val HIVE_METASTORE_PARTITION_PRUNING_FALLBACK =
+    buildConf("spark.sql.hive.metastorePartitionPruningFallback")
```

Comment:
Should we use `spark.sql.legacy` prefix like [SPARK-19724](https://github.com/apache/spark/pull/22515), @rxin and @cloud-fan ?

Reply:
What is the reasoning for marking this as legacy?

Reply:
spark.sql.hive.metastorePartitionPruning.fallback.enabled